src/Package_Command.php

```diff
@@ -952,6 +950,37 @@ private function is_git_repository( $package ) {
 	}
 
 	/**
+	 * Check that `$package_name` matches the name in the repo composer.json, and return corrected value if not.
+	 */
+	private function check_git_package_name( $package_name ) {
+		// Generate raw git URL of composer.json file.
+		$raw_content_url = 'https://raw.githubusercontent.com/' . $package_name . '/master/composer.json';
```

Comment:
I think we still have a problem here.
The branch `master` is hardcoded, so the code will pull in the wrong `composer.json` file if a different branch was requested.
Not a bug with this PR, but rather a bug we already had in the previous iteration.

Reply:
Ah right so `check_git_package_name()` needs to take a `$version` arg as well - should this be a new issue/PR or deal with it here?

Reply:
I'd say a new PR, as the bug is already in there. This way, we can concentrate on the issue at hand first, and deal with the other bug separately.

Reply:
I'll merge this one and create a new issue for that bug.

Reply:
See https://github.com/wp-cli/package-command/issues/51